app/models/create_cap_proposal.rb

```diff
@@ -0,0 +1,90 @@
+class CreateCapProposal
+  def initialize(auction)
+    if !auction.is_a?(Presenter::Auction)
+      fail "an instance of Auction::Presenter must be passed to CreateCapProposal::initialize"
+    end
+
+    @auction = auction
+  end
+
+  def configure_c2_client!
+    @host = ENV.fetch('C2_HOST', 'https://cap.18f.gov')
+    @c2_client = C2::Client.new(
+      oauth_key: ENV.fetch('C2_OAUTH_KEY'),
+      oauth_secret: ENV.fetch('C2_OAUTH_SECRET'),
+      host: @host,
+      debug: ENV.fetch('C2_DEBUG', false)
+    )
+
+    @c2_client
+  end
+
+  def c2_client
+    @c2_client ||= configure_c2_client!
+  end
+
+  def perform
+    resp = @c2_client.post('proposals', c2_proposal_hash)
+    proposal = resp.body
+
+    construct_proposal_url(proposal.id)
+  end
+
+  def c2_proposal_hash
+    {
+      office: 'DC',
+      purchase_type: 'Software',
+      product_name_and_description: product_name_and_description,
+      justification: justification,
+      link_to_product: link_to_product,
+      cost_per_unit: cost_per_unit,
+      quantity: 1,
+      recurring: false,
+      date_requested: date_requested,
+      urgency: 20,
```

Comment:
can we put this in a constant to reveal naming? I think we could use enum value, too :) (hashtag inside knowledge of C2)